docs(api): add `<NuxtLayout>` component docs (#6264)
nuxt/framework: docs/content/3.api/2.components/2.nuxt-layout.md

# `<NuxtLayout>`

> You can use `<NuxtLayout/>` component to activate `default` layout on `app.vue` or `error.vue` .

Reviewer: ```suggestion
You can use `<NuxtLayout />` component to activate `default` layout on `app.vue` or `error.vue`.
```

```vue [/app.vue]
<template>
  <NuxtLayout>
    some page content
  </NuxtLayout>
</template>
```

> `<NuxtLayout />` is also used to override `default` layout on `app.vue` , `error.vue` or even page components found in the `/pages` directory.

Reviewer: ```suggestion
`<NuxtLayout />` can also be used to override `default` layout on `app.vue`, `error.vue` or even page components found in the `/pages` directory.
```

> ## `name` props

Reviewer: ```suggestion
## `name` prop
```

> `<NuxtLayout/>` component accepts `name` property to pass the name of the custom layout. You can pass `name` props to override the default layout, where `name` can be a static string, reactive reference or even a computed property as long as it matches the name of the layout file in `/layouts` directory.

Reviewer: ```suggestion
`<NuxtLayout />` component accepts the `name` prop, which you can pass to use a non-default layout, where `name` can be a static string, reactive reference or a computed property. It **must** match the name of the corresponding layout file in the `/layouts` directory.
```

### Examples

```vue [pages/index.vue]
<template>
  <NuxtLayout :name="layout">
    <NuxtPage />
  </NuxtLayout>
</template>

<script setup>
// layouts/custom.vue

> const layout = "custom";

Reviewer: ```suggestion
const layout = 'custom'
```

</script>
```

> The layout name is normalized to kebab-case, so if your layout file is named, `errorLayout.vue` , it will become `error-layout` when passing as a `name` property to `<NuxtLayout />`.
> 
> ```vue [/error.vue]
> <template>
>   <NuxtLayout :name="error-layout">
>     <NuxtPage />
>   </NuxtLayout>
> </template>
> ```

Reviewer: ```suggestion
::alert{icon=👉}
Please note the layout name is normalized to kebab-case, so if your layout file is named `errorLayout.vue`, it will become `error-layout` when passed as a `name` property to `<NuxtLayout />`.

```vue [/error.vue]
<template>
  <NuxtLayout :name="error-layout">
    <NuxtPage />
  </NuxtLayout>
</template>
```
::
```

## Layout and transition

> `<NuxtLayout />` renders incoming content via `<slots />`. This incoming content is then wrapped around Vue’s `Transition` component to activate layout transition. For this layout transition to work as expected, it is recommended that `<NuxtLayout />` is **not** the root element of the page component.

Reviewer: ```suggestion
`<NuxtLayout />` renders incoming content via `<slot />`, which is then wrapped around Vue’s `<Transition />` component to activate layout transition. For this to work as expected, it is recommended that `<NuxtLayout />` is **not** the root element of the page component.
```
